include/deal.II/lac/precondition.h

```diff
@@ -2397,7 +2397,14 @@ namespace internal
                   const unsigned int        n_iterations)
   {
     typename VectorType::value_type eigenvalue_estimate = 0.;
-    eigenvector /= eigenvector.l2_norm();
+
+    const auto norm = eigenvector.l2_norm();
+
+    if (norm == 0.0)
+      return 1.0;
+
+    eigenvector /= norm;
```

Comment:
I think that's the wrong solution. If you pass in a zero vector by accident, the function pretends that the largest eigenvalue of the operator is one. That's just not true.

The function should probably document that one is must provide a non-zero vector and that if the vector *is* zero throw an exception that one can catch.